### ISSUE TITLE
[RFC][Messenger] Remove ResetLoggersWorkerSubscriber

### ISSUE THREAD

**Issue by lyrixx** [symfony/symfony#43271]
Hello, 

Since #41163 and #43133 I think #40761 is not needed anymore.

**TL;DR:**

* #41163 In Symfony 5.4 a new system could reset the container between each (AMQP, redis, etc) message. This works by using ServiceResetter between each message (in a worker only)
* #43133 In symfony 5.4, a deprecation notice appear to force user to enable `reset_on_message` (disabled by default to be BC). 
* In symfony 6.0 the value will be true
* #40761 Introduce a way to reset monolog between each message
* Everything has been introduced in SF 5.4 (not released, so it's now or never)

=> The first feature embrace the the last one, and will be shipped by default.

That's why I propose to remove all code related to #40761.

ping @upyx @okwinza 

@l-vo I'm deeply sorry for trying to remove your code. I know it's really unpleasant when someone want to remove our code. But in this case, I think it's legit. It simplifies the Symfony code base, project configuration, and documentation.

---

If everyone is OK, I'll work on a PR.

**Comment by fabpot** [comment]
#40761 has been introduced in 5.3, not 5.4, so we need a proper deprecation in 5.4 and full removal on 6.0.
:+1: for doing so.
